### PR TITLE
GCP disconnected & private, with workload identity

### PR DESCRIPTION
**Pull request by jianli-wei**
1. 4.12 epic [CORS-2361 Deploy disconnected OpenShift cluster with Workload Identity on Google Cloud](https://issues.redhat.com/browse/CORS-2361)
2. add the workflow `cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity` and provision & deprovision chains
3. update the step `gather-gcp-console` to deal with private clusters

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/deprovision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-deprovision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+chain:
+  as: cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-deprovision
+  steps:
+  - ref: gather-mirror-registry
+  - chain: cucushift-installer-rehearse-gcp-ipi-deprovision
+  - ref: ipi-conf-gcp-oidc-creds-deprovision
+  - ref: gcp-deprovision-bastionhost
+  - ref: gcp-deprovision-vpc
+  documentation: |-
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/deprovision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-deprovision-chain.yaml:9]: How about like this order:
```
- chain: cucushift-installer-rehearse-gcp-ipi-disconnected-deprovision
- ref: ipi-conf-gcp-oidc-creds-deprovision
```

**Comment by jianlinliu** [comment]
In https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/33459/rehearse-33459-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp/1585195667494539264, it failed on `gcp-gather-gcp-console`, can you help fix it like https://github.com/openshift/release/blob/master/ci-operator/step-registry/gather/aws-console/gather-aws-console-commands.sh#L17-L21

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - ref: gcp-provision-vpc
+  - ref: ignition-bastionhost
+  - ref: gcp-provision-bastionhost
+  - chain: mirror-images-by-oc-adm
+  - ref: proxy-config-generate
+  - ref: ipi-conf
+  - ref: ipi-conf-telemetry
+  - ref: ipi-conf-gcp
+  - ref: ipi-conf-manual-creds
+  - ref: ipi-conf-gcp-oidc-creds-provision
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-provision-chain.yaml:14]: For this workflow, maybe we can wait for https://github.com/openshift/release/pull/33641/files get merged, then insert `ipi-conf-manual-creds-remove-unnecessary-creds` here too.

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-provision-chain.yaml:14]: @jianlinliu Do you think I shall move the updates on step `gather-gcp-console` (which triggered many jobs) to another PR, and then wait for #33641getting merged in this PR?

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-provision-chain.yaml:14]: I saw you already updated `gather-gcp-console` step in this PR. Do you mean will insert `ipi-conf-manual-creds-remove-unnecessary-creds` in another PR?

Comment (jianli-wei) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/private/cco-manual-workload-identity/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-private-cco-manual-workload-identity-provision-chain.yaml:14]: I meant to revert the changes on `gather-gcp-console` and wait for #33641 getting merged. But as #33641 had been merged, so please ignore my comment. Just inserted `ipi-conf-manual-creds-remove-unnecessary-creds`.

**Comment by jianli-wei** [comment]
Analyzed the above failures, none was caused by the PR. 

Test name | Comments
-- | -- 
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.11-gcp-ipi-cco-manual-workload-identity-p1 | INFO[2022-11-03T04:36:06Z] Step phase pre succeeded after 41m21s.
ci/prow/multi-arch-gen-valid | GIT Automatic merge failed
ci/rehearse/openshift/origin/master/e2e-gcp-ovn | INFO[2022-11-03T04:51:11Z] Step phase pre succeeded after 41m20s.
ci/rehearse/openshift/origin/release-4.6/e2e-agnostic-cmd | INFO[2022-11-03T04:44:08Z] Step phase pre succeeded after 40m20s.
ci/rehearse/openshift/origin/master/e2e-gcp-image-ecosystem | RRO[2022-11-03T04:07:49Z] could not run steps: step [release:latest] failed: release "release-latest" failed: the pod ci-op-hj71ft8v/release-latest failed after 59s (failed containers: release): ContainerFailed one or more containers exited 
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.11-gcp-ipi-ovn-p1 | INFO[2022-11-03T04:34:26Z] Step phase pre succeeded after 39m20s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.11-gcp-ipi-ovn-ipsec-p1 | INFO[2022-11-03T04:36:15Z] Step phase pre succeeded after 41m30s.
ci/rehearse/openshift/gcp-filestore-csi-driver-operator/main/operator-e2e | INFO[2022-11-03T04:59:38Z] Step phase pre succeeded after 46m11s.
ci/rehearse/operator-framework/operator-marketplace/master/e2e-gcp-serial | INFO[2022-11-07T07:51:47Z] Step phase pre succeeded after 44m0s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-gcp-ipi-disconnected-private-ovn-p2 | INFO[2022-11-07T08:10:36Z] Step phase pre succeeded after 1h11m11s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-gcp-ipi-cco-manual-workload-identity-p1 | INFO[2022-11-07T07:55:05Z] Step phase pre succeeded after 55m31s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-gcp-ipi-baselinecaps-v411-additionalcaps-p2 | INFO[2022-11-07T07:49:13Z] Step phase pre succeeded after 50m30s.
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-gcp-ipi-disconnected-p2 | INFO[2022-11-07T08:17:56Z] Step phase pre succeeded after 1h18m31s.
ci/rehearse/openshift/cloud-credential-operator/master/e2e-gcp-manual-oidc | 2022/11/07 07:11:23 Failed to create IAM service accounts: Failed while processing each CredentialsRequest: Failed to add predefined roles for IAM service account ci-op-7ndlflyk-edea3-openshift-image-registry-gcs: error setting project policy: googleapi: Error 409: There were concurrent policy changes. Please retry the whole read-modify-write with exponential backoff. The request's ETag '\007\005\354\334!X\024+' did not match the current policy's ETag '\007\005\354\334!\204\350\032'., aborted

File: ci-operator/step-registry/gather/gcp-console/gather-gcp-console-commands.sh [hunk 3]
```diff
@@ -17,6 +17,18 @@ fi
 
 gcloud config set project "$(jq -r .gcp.projectID "${SHARED_DIR}/metadata.json")"
 
+# while gathering logs from a private cluster, proxy setting is required for connecting cluster
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	echo "Going to enable client proxy by 'source ${SHARED_DIR}/proxy-conf.sh'"
+	echo "------------------------------"
+	cat "${SHARED_DIR}/proxy-conf.sh"
```
Comment (jianlinliu) [inline ci-operator/step-registry/gather/gcp-console/gather-gcp-console-commands.sh:25]: The content is a bit sensitive, so it is better not print them out.

Comment (jianli-wei) [reply ci-operator/step-registry/gather/gcp-console/gather-gcp-console-commands.sh:25]: I don't think the printed info is sensitive (see below). BTW, the main concern for me is, if any further commits, many jobs would be triggered another round and I have to analyze their failures once more. Please advise, thanks! 

FYI https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/33459/rehearse-33459-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp/1589512086084390912/artifacts/installer-rehearse-gcp/gather-gcp-console/build-log.txt
Going to enable client proxy by 'source /tmp/secret/proxy-conf.sh'
------------------------------
export http_proxy=http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@35.238.191.86:3128
export https_proxy=http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@35.238.191.86:3128
export no_proxy="localhost,127.0.0.1"
export HTTP_PROXY=http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@35.238.191.86:3128
export HTTPS_PROXY=http://XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@35.238.191.86:3128
export NO_PROXY="localhost,127.0.0.1"
------------------------------

**Pull request merged**